tests/post/steps/test_volume.py

```diff
@@ -153,6 +157,49 @@ def delete_pvc(host, name, k8s_client):
         grace_period_seconds=0
     )
 
+
+@when(parsers.parse("I delete the pod '{name}'"))
+def delete_namespaced_pod(host, name, k8s_client):
+    # ensure pod is available before deleting
```

Comment:
No.

Pods are created with a `given`, the `given` should wait for availability. Not the delete.

Reply:
And for the wait on a pod, you should reuse `kube_utils.wait_for_pod`.

Reply:
@slaperche-scality, i think i implemented the wait here because in the complete E2E test workflow, when we create the pod, it cannot be in available state because it has a pvc which is not yet fulfilled since the volume is only created in the `when`

Reply:
Thinking we should create the volume in the `given` to furfill this check